ttnn/cpp/ttnn/operations/embedding/device/embedding_device_operation.cpp

```diff
@@ -56,9 +56,9 @@ void Embeddings::validate(const std::vector<Tensor> &input_tensors) const {
 std::vector<TensorSpec> Embeddings::compute_output_specs(const std::vector<Tensor> &input_tensors) const {
     const auto &input_tensor = input_tensors.at(0);
     const auto &weight_tensor = input_tensors.at(1);
-    auto num_output_embeddings = input_tensor.logical_shape()[3];
+    auto num_output_embeddings = input_tensor.logical_shape()[-1];
     auto batch_num = input_tensor.logical_shape()[0];
-    auto num_embedding_dims = weight_tensor.logical_shape()[3];
+    auto num_embedding_dims = weight_tensor.logical_shape()[-1];
```

Comment:
This was from a previous change but careful when using `logical_shape`.  

`logical_shape()` is the async getter, which is fairly untested and `get_logical_shape()` is the standard synchronous one. Clearly seems like it hasn't caused bugs here but it could be problematic.